src/cow-react/utils/getIntOrFloat.ts

```diff
@@ -0,0 +1,9 @@
+// Allow only int or float
+// Examples: 5 or 5.4 or 1. or 0.333 or .12
+export function getIntOrFloat(amount: string | null): string | null {
+  if (!amount) return null
+
+  if (/^\.\d+|\d+\.?\d*$/.test(amount)) return amount
```

Comment:
Probably `/^\d+\.?\d*$/`  is enough.

Otherwise, this seems simpler:
`/^\d+(?:\.\d*)?$`

Reply:
aha! makes sense!